### PR TITLE
Make ppx_const compatible with ppxlib.0.18.0

### PR DESCRIPTION
**Pull request by NathanReb**
ppxlib.0.18.0 updates its internal AST to 4.11 resulting in a change in string constant representation. This PR makes ppx_const compatible with the latest ppxlib.

You will probably want to wait for the ppxlib release to go through before merging this!

### CODE REVIEW

File: ppx_const.opam [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # This file is generated by dune, edit dune-project instead
 opam-version: "2.0"
 version: "2.0"
-synopsis: "Compile-time \"if\" statement for conditional inclusion of code"
+synopsis: "Compile-time \"if\" statement for conditional inclusion of code."
 description: """
 This is a ppx extension which adds `if#const` and `match#const` constructs to
-OCaml. They behave like normal `if` and `match`, but conditions are evaluated
+OCaml. They behave like normal `if` and `const`, but conditions are evaluated
```
Comment (mcclure) [inline ppx_const.opam]: Your two changes at the top of ppx_const.opam (the removed period and the substitution of "match" for "const") are incorrect. These are typos that were fixed in the most recent release, you have reintroduced them. Please revert these changes.

File: src/ppx_const.ml [hunk 2]
```diff
@@ -25,7 +25,7 @@ let const_eq x y = match x, y with
       String.equal x y
   | Pconst_char x, Pconst_char y ->
       Char.equal x y
-  | Pconst_string (x, _delim1), Pconst_string (y, _delim2) ->
+  | Pconst_string (x, _, _delim1), Pconst_string (y, _, _delim2) ->
```
Comment (mcclure) [inline src/ppx_const.ml:28]: Can you confirm that this is only necessary for ppxlib 0.9.0->0.18.0 compatibility, and won't break compatibility with older versions of ocaml?

**Comment by mcclure** [comment]
Thank you, I've been out of the OCaml community for some time and I'm a little confused-- can you confirm, this PR updates for compatibility with an *upcoming* ppxlib release?

Can you come back and let me know when ppxlib 0.18.0 is released, alternately is there a Github issue or something I can subscribe to to be notified?

File: ppx_const.opam (hunk 1 above)
Comment (NathanReb) [reply ppx_const.opam]: Hmm I was first surprised to see those changes as I didn't rewrite these parts myself. It turns out your `dune-project` file's package description is desynchronized from your opam file.

I think the proper fix would be to update your `dune-project` instead, I'll take care of it!

File: src/ppx_const.ml (hunk 2 above)
Comment (NathanReb) [reply src/ppx_const.ml:28]: Indeed, ppxlib will take care of ensuring compat with older OCaml versions + you gain 4.11 compatibility here.

**Comment by NathanReb** [comment]
> Can you come back and let me know when ppxlib 0.18.0 is released, alternately is there a Github issue or something I can subscribe to to be notified?

Of course, I'll send a message here as soon as ppxlib.0.18.0 is released!

**Comment by mcclure** [comment]
@NathanReb thanks for the explanation on the dune file, whenever you let me know 0.18.0 is available I will merge and update the package on opam

**Comment by NathanReb** [comment]
ppxlib.0.18.0 has been released so this should be good to go! Could you restart the CI builds just to make sure it all works out please?

**Comment by mcclure** [comment]
This is a new build I booted 2 hours ago

https://travis-ci.org/github/mcclure/ppx_const/builds/729535747

and it's still failing. ppxlib seems to be the problem

```
The following dependencies couldn't be met:
  - ppx_const → ppx_getenv >= 2.0 → ppxlib < 0.18.0
  - ppx_const → ppxlib >= 0.18.0
Your request can't be satisfied:
  - No available version of ppxlib satisfies the constraints
```

**Comment by mcclure** [comment]
@NathanReb, I started a new build to see if something just hadn't propagated yet when I started the last build, but it is failing with the same error. It seems something is wrong with the PR.

I will have time to do the release this weekend if you can get me a fixed PR by then, if not, it may have to wait until next weekend. Thanks

**Comment by kit-ty-kate** [comment]
@mcclure a compatible `ppx_getenv` (2.1) was released two weeks ago. It should work now if you restart the CI.

**Comment by kit-ty-kate** [comment]
Just so you know, this PR is required to add the OCaml 4.12 support to ppx_const.

**Comment by mcclure** [comment]
Kate, thanks for reminding me, I restarted the build when you made your comment 12 days ago and then I forgot to actually come back and merge…! The bot claims the CI failed but the most recent one succeeded.
